system/Autoloader/FileLocator.php

```diff
@@ -83,27 +73,26 @@ public function __construct(Autoload $autoload)
 	 * @param string $folder The folder within the namespace that we should look for the file.
 	 * @param string $ext    The file extension the file should have.
 	 *
-	 * @return string       The path to the file if found, or an empty string.
+	 * @return string|false The path to the file, or false if not found.
 	 */
-	public function locateFile(string $file, string $folder = null, string $ext = 'php'): string
```

Comment:
Yes, it works with `false`. Everyplace has an `if (empty(...))` after call this method.

Really, it might not be necessary to return false if the file does not exist. I preferred to update thinking about making it clearer when the file was not found returning false and not a string.

Several Filesystem Functions return false on failure, eg: [realpath](http://php.net/manual/en/function.realpath.php#refsect1-function.realpath-returnvalues)